src/main/java/uk/gov/hmcts/reform/jobscheduler/jobs/HttpCallJob.java

```diff
@@ -19,10 +21,16 @@
 
     private final RestTemplate restTemplate;
     private final ActionExtractor actionExtractor;
+    private final AuthTokenGenerator tokenGenerator;
 
-    public HttpCallJob(RestTemplate restTemplate, ActionExtractor actionExtractor) {
+    public HttpCallJob(
+        RestTemplate restTemplate,
+        ActionExtractor actionExtractor,
+        @Qualifier("cachedServiceAuthTokenGenerator") AuthTokenGenerator tokenGenerator
```

Comment:
What if the token is no longer valid? We'll wait for 4h until a new is retrieved by this generator.
Perhaps on 401/403 response from the target server we could retrieve fresh token and try again?

Reply:
I can see two options:
1. lowering ttl in configuration
2. leaving it to the retry mechanism (from another story)

Reply:
Given how CachedServiceAuthTokenGenerator is implemented (no resetting), I'd just lower ttl for now, so that another attempt to execute the same job (after its failure) would use a different token. As Rafał mentioned to me, some improvements can probably be done in the generator's code. This way we (and everyone else) wouldn't have to worry about caching expired tokens.